### DISCUSSION TITLE
Airflow sending negative values

### DISCUSSION THREAD

**Discussion by regisferlima** [apache/airflow#29787] [General]
Hi,
we having a problem with this dag: _airflow.dagrun.duration.success.avg_

We sent the logs to Datadog but they said that this dag is sending negative values to Datadog.

Answer from datadog:

_Unfortunately there's not much that I can see which indicates why this metric value is reporting negative. Since the trace level flare showed that the Agent is receiving a negative value, that indicates to me that the metric may be sent to Datadog as a negative value.

Thank you,
Janae Quinones | Solutions Engineer | Datadog_

Do you know guys how we can see if it is working correctly? how can we see the values inside the airflow, I believe it is related to StatsD.

**Comment by potiuk** [comment]
Might be a bug. Please create an issue describing all the details, logs, what is logged via statsd, what are your deployment is. 

But my very wild guess is that you have no time synchronization in your machines and your time drifts between different machine by a number of seconds. Then reported duration might be negative. Whenever you run any distributed system (airflow is not an exception) you have to make sure all the machines/containers you run have time synchronized down to microseconds (for example using ntpd). It's normal tha time of running system drifts (sometimes by seonds per day) and ntpd or similar will make sure to remove the drift.
